src/avformat/avio.rs

```diff
@@ -154,6 +154,12 @@ impl AVIOContextCustom {
             _opaque: opaque,
         }
     }
+
+    // Extract the data passed in `alloc_context`
+    // and set that data inside it to an empty vec.
+    pub fn take_data(&mut self) -> Vec<u8> {
```

Comment:
People might want to manipulate more on the data, I think we can provide a method such as `fn as_mut_data(&mut self) -> &mut Vec<u8>` here.